src/app/modules/twenty-four-hour-time/twenty-four-hour-time.component.html

```diff
@@ -0,0 +1,168 @@
+<label class="ds-c-label" [ngClass]="labelClass" for="{{ id }}">
+  <span class="ds-u-font-weight--bold">{{ title }}</span>
+  <span *ngIf="hint" class="ds-c-field__hint" [ngClass]="hintClass" role="alert">{{ hint }}</span>
+  <span *ngIf="errorMessage" class="ds-c-field__hint ds-u-color--error" role="alert">{{ errorMessage }}</span>
+</label>
+
+<ng-template [ngIf]="alertMessageList?.length">
+  <div class="ds-u-margin-y--2">
+    <app-alert dataAutoId="{{ id + '-' + alertVariation }}" [hideClose]="true" variation="{{ alertVariation }}">
+      <div *ngIf="alertMessageList.length > 1; else singleAlert">
+        <div *ngFor="let alertMessage of alertMessageList">
+          <li>{{ alertMessage }}</li>
+        </div>
+      </div>
+      <ng-template #singleAlert>
+        {{ alertMessageList[0] }}
+      </ng-template>
+    </app-alert>
+  </div>  
+</ng-template>
+
+<div aria-live="assertive" class="sr-only">
+  <div *ngIf="screenReaderAnnouncement.length > 0">{{screenReaderAnnouncement}}</div>
+</div>
+
+<div
+  id="time-container-{{id}}"
+  [ngClass]="inputClass + ' ' + 
+    (errorMessage ? 'time-container-error-border' : 'time-container-border') + ' ' +
+    (disabled ? 'disabled-background' : 'enabled-background')"
+  class="time-container
+    ds-l-row
+    ds-u-margin-y--1"
+  >
+  <div class="ds-l-col-10 input-padding">
+    <input
+      class="time-input ds-u-font-size--sm"
+      name="{{ id }}"
+      attr.aria-label="{{ ariaLabel || title }}"
+      attr.data-auto-id="{{ dataAutoId }}"
+      type="text"
+      tabindex="0"
+      onClick="this.setSelectionRange(0, this.value.length)"
+      maxlength="5"
+      [disabled]="disabled"
+      [formControl]="control"
+      [formlyAttributes]="formlyAttributes"
+      (input)="validateInput($event)"
+      (change)="formatAndSetTime($event)"
+      (blur)="blurCallBackFunction($event)"
+      (keyup)="keyupCallBackFunction($event)"
+      (focus)="addFocusStyling()"
+    />
+  </div>
+  <div class="ds-l-col--2 clock-parent">
+    <span
+      role="button"
+      tabindex="0"
+      alt="clock"
+      class="clock-image"
+      aria-label="Activate enter to open the time picker."
+      (click)="openTimePicker($event)"
+      (keyup.enter)="openTimePicker($event)"
+    ></span>
+  </div>
+</div>
+<div class="time-picker
+    ds-u-radius
+    ds-u-border--1
+    ds-u-border--dark"
+  *ngIf="timePickerVisible"
+  id="time-picker-{{id}}"
+  (click)="stopPropagation($event)"
+  (focusout)="focusLost($event)"
+>
+  <div class="time-picker-header">
```

Comment:
Any classes I could find from CMS design system around
- padding, margins, layout etc
- visual text size, style etc
... I try to keep in the HTML, rather than re-create in CSS

Although maybe we already have enough inconsistencies that there's not much payoff in being that strict?

But, functionally, or at least clarity/readability, I find I like to keep the padding/spacing styles in the HTML. It helps me keep the effect of all these straight and read them alongside the markup, since I'm usually trying to visualize it alongside the pile of `<div>` as I work.

Then any of the `flex` or further styling I put in "semantically" named CSS styles. I see you're also doing this, and it really helps. I just noticed more padding/margin and text style control in your CSS.

Not really a _request_ for change in here, just something I noticed. Since we're constantly discovering and settling on conventions, and I happened to be working on a new component, figured we could chat about whether it's even worth a convention.